src/services/resolver/GithubAliasResolver.php

```diff
@@ -99,17 +100,17 @@ private function initRateLimit(): void {
         $this->rateLimit = $response->getRateLimit();
     }
 
-    private function updateRateLimit(): void {
+    private function updateRateLimit(RateLimit $previousRateLimit): void {
         if ($this->fileDownloader->hasRateLimit()) {
             $this->rateLimit = $this->fileDownloader->getRateLimit();
 
             return;
         }
 
         $this->rateLimit = new RateLimit(
-            $this->rateLimit->getLimit(),
-            $this->rateLimit->getRemaining() - 1,
-            $this->rateLimit->getResetTime()
+            $previousRateLimit->getLimit(),
+            $previousRateLimit->getRemaining() - 1,
+            $previousRateLimit->getResetTime()
```

Comment:
Why does passing this in instead of using the existing property make anything better? ;)

Reply:
See my previous comment.
It guarantees that `getLimit()`, `getRemaining()` and `getResetTime()` won't be called on null.